clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -711,6 +711,7 @@ public void onResponse(final ClientResponse response) {
             long currentTimeMs = response.receivedTimeMs();
             OffsetCommitResponse commitResponse = (OffsetCommitResponse) response.responseBody();
             Set<String> unauthorizedTopics = new HashSet<>();
+            boolean hasFailedAttempt = false;
```

Comment:
nice. I would only suggest renaming it to show that this is about recording/registering the failure for the request.  "failedRequestRegistered" maybe? WDYT?

Reply:
Great idea. It's more clear than mine.